cmd/http/listener.go

```diff
@@ -73,11 +73,12 @@ func isHTTPMethod(s string) bool {
 	return false
 }
 
-func getResourceHost(bufConn *BufConn, maxHeaderBytes, methodLen int) (resource string, host string, err error) {
+func getResourceHost(bufConn *BufConn, maxHeaderBytes, methodLen int) (resource string, host string) {
 	var data []byte
-	for dataLen := 0; methodLen+dataLen < maxHeaderBytes; dataLen += 8 {
-		if data, err = bufConn.Peek(methodLen + dataLen); err != nil {
-			return "", "", err
+	for dataLen := 0; methodLen+dataLen < maxHeaderBytes; dataLen += 4 {
```

Comment:
Reducing from `8` to `4` does not solve this problem.  The actual problem is `bufConn.Peek()` infinitely waits to completely peek into given length.

Right way to fix is to have custom implementation of peek specifically to `bufConn`

Reply:
As I mentioned earlier it does not fix the problem.  For below HTTP request, it will hang because of infinite wait because peek waits for requires three more bytes (methdLen=4, request contains 37 bytes) to complete whereas there is no more data on the TCP connection.

```
GET / HTTP/1.1\r\nHost: example.org\r\n\r\n
```